code/modules/admin/verbs/getlogs.dm

```diff
@@ -1,41 +1,63 @@
-/*
-	HOW DO I LOG RUNTIMES?
-	Firstly, start dreamdeamon if it isn't already running. Then select "world>Log Session" (or press the F3 key)
-	navigate the popup window to the data/logs/runtime/ folder from where your tgstation .dmb is located.
-	(you may have to make this folder yourself)
-
-	OPTIONAL: 	you can select the little checkbox down the bottom to make dreamdeamon save the log everytime you
-				start a world. Just remember to repeat these steps with a new name when you update to a new revision!
-
-	Save it with the name of the revision your server uses (e.g. r3459.txt).
-	Game Masters will now be able to grant access any runtime logs you have archived this way!
-	This will allow us to gather information on bugs across multiple servers and make maintaining the TG
-	codebase for the entire /TG/station commuity a TONNE easier :3 Thanks for your help!
-*/
-
-
-//This proc allows download of past server logs saved within the data/logs/ folder.
+/// This proc allows download of past server logs saved within the data/logs/ folder.
 /client/proc/getserverlogs()
 	set name = "Get Server Logs"
 	set desc = "View/retrieve logfiles."
 	set category = "Admin"
 
-	var/path = browse_files("data/logs/")
-	if(!path)
+	access_file_by_browsing_path(usr, "data/logs/")
+
+/// This proc allows download of past server logs saved within the data/logs/ folder by specifying a specific round ID.
+/client/proc/get_server_logs_by_round_id()
+	set name = "Get Round Logs"
+	set desc = "View/retrieve logfiles for a given round."
+	set category = "Admin"
+
+	var/round_id = input(usr, "Enter a round ID.", "Enter Round ID", "[GLOB.round_id]") as null|text
+	if(!round_id)
+		return
+
+	var/round_path = "[GLOB.log_directory]/"
+	if(round_id != "[GLOB.round_id]")
+		var/datum/db_query/query = SSdbcore.NewQuery(
+			"SELECT UNIX_TIMESTAMP(initialize_datetime) AS dt FROM round WHERE id = :id",
+			list(id = round_id)
+		)
+		if(!query.warn_execute())
+			qdel(query)
+			to_chat(usr, "Could not check database for round [round_id].")
+			return
+
+		if(!query.NextRow())
+			qdel(query)
+			to_chat(usr, "Could not find round [round_id] in database.")
+			return
+
+		// convert unix timestamp in seconds to byond timestamp in deciseconds
+		var/round_datetime = (query.item[1] - BYOND_EPOCH_UNIX) * 10
```

Comment:
Yo this is a thing????

Reply:
not sure which part of this you're referring to; world.timeofday is in deciseconds since the byond epoch, 1/1/2000, so i added this constant to convert between the two. it's worked for me locally. there might be some weirdness around the day boundary/midnight but it should work just fine for the vast vast majority of our rounds and the directory where their logs end up.